pkg/controller/druid/handler.go

```diff
@@ -879,6 +926,28 @@ func namespacedName(name, namespace string) *types.NamespacedName {
 	return &types.NamespacedName{Name: name, Namespace: namespace}
 }
 
+// IntPointer returns a pointer to int32.
+func IntPointer(i int32) *int32 {
+	return &i
+}
+
+// GetHPAReplicaCountOrDefault returns the count of pods in case hpa is mentioned.
+func GetHPAReplicaCountOrDefault(name types.NamespacedName, defaultReplicaCount int32) int32 {
+	var hpa autoscalev2beta1.HorizontalPodAutoscaler
```

Comment:
this doesn't appear to be initialized anywhere  according to what we have in CR ?

Reply:
dint get you. its just initialized in this function

Reply:
this line initializes it to empty HorizontalPodAutoscaler struct , and then it is directly used at line 944 , we don't seem to be reading the actual value that user might have put in the CR or  are you just trying to set to default min value of empty `HorizontalPodAutoscaler` for which `hpa.Spec.MinReplicas` is always nil and `hpa.Status.DesiredReplicas` is always 0.

Reply:
Yes, replica will show different than minreplicas i guess thats what we agreed on. Plus hpa.Sepc.MinReplicas will show what is mentioned not zero. Here's a snippet 
```
"apiVersion\":\"apps/v1\"},\"minReplicas\":1,\"maxReplicas\":10,\"metrics\":[{\"type\":\"Resource\",\"resource\":{\"name\":\"cpu\",\"targetAverageUtilization\":60}}]},\"status\":{\"currentReplicas\":0,\"desiredReplicas\":0,\"currentMetrics\":null,\"conditions\":null}}","name":"tiny-cluster","namespace":"default"}
```
Plus to clarify this way the whole pod creation takes place, the stateful set is created with replica 1 ( assume we mention replica 1) , replica one get created, HPA get deployed ( assume we mention 3 as minReplica ), after seeing the desired to replica to 3. Hpa takes replica to count 3. 
Earlier with client, in case hpa mentioned it would directly take count to 3.

Reply:
Seems this function is really making no sense. This would only make sense in case ```replicas *int32```, since replica is required and is absolute no point having this. I guess its overdo.  I get it this let me muke this. Lets keep it straightforward. @himanshug

Reply:
Yeah, I would prefer to just  let the StatefulSet be as it was before and not try to adjust replica count on statefulSet at all.